homeworks/yuriy.chaban_YuriyChaban/homework_5/task_4.js

```diff
@@ -0,0 +1,60 @@
+const people = [
+    {
+        id: 1,
+        name: 'Nick',
+        friends: [2, 5, 6],
+    },
+    {
+        id: 2,
+        name: 'John',
+        friends: [1, 3],
+    },
+    {
+        id: 3,
+        name: 'Mike',
+        friends: [2, 5],
+    },
+    {
+        id: 4,
+        name: 'Janny',
+        friends: null,
+        // friends: [7, 9],
+    },
+    {
+        id: 5,
+        name: 'Edward',
+        friends: [1, 3],
+    },
+    {
+        id: 6,
+        name: 'Jeen',
+        friends: [5, 1],
+    },
+];
+
+const getPeople = (userId, arrayOfUsers) => {
+    const friendsArray = [];
+    arrayOfUsers.forEach((obj) => {
+        if (obj.friends === null) {
+            return;
+        }
+        obj.friends.forEach((friendId) => {
+            if (friendId === null) {
+                return;
+            }
+
+            friendsArray.push(friendId);
+
+            if (userId > Math.max(friendsArray)) {
+                console.log('The Id of friends, that you want to get does not exist in the array of friends!');
+            }
```

Comment:
is this really the proper way to check if the item is in array?

Reply:
two simple steps:

1. `find()` user in array by id
2. `filter()` the original array by users friends ids

that's it

Reply:
markoff den, I tried to use find and filter, and this is doesn't help me, because it was a iteration of all items in array and output into console result for every iteration if false. Could you please add more information how it will be?